server/config/store_config.go

```diff
@@ -118,40 +95,115 @@ func (c *StoreConfig) GetRegionMaxKeys() uint64 {
 	return uint64(c.Coprocessor.RegionMaxKeys)
 }
 
-// UpdateConfig updates the config with given config map.
-func (m *StoreConfigManager) UpdateConfig(c *StoreConfig) {
-	if c == nil || m == nil {
-		return
+// StoreConfigManager is used to manage the store config.
+type StoreConfigManager struct {
+	config atomic.Value
+	source Source
+}
+
+// NewStoreConfigManager creates a new StoreConfigManager.
+func NewStoreConfigManager(client *http.Client) *StoreConfigManager {
+	schema := "http"
+	if ts, ok := client.Transport.(*http.Transport); ok && ts.TLSClientConfig != nil && len(ts.TLSClientConfig.Certificates) > 0 {
```

Comment:
Is it enough to make sure whether TLS is enabled in this way?

Reply:
I will test it in tls enviroment.

Reply:
fix